cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -12,21 +12,50 @@
 pytest_generate_tests = testgen.generate(testgen.provider_by_type, ["hawkular"], scope="function")
 
 
+@pytest.mark.usefixtures('has_no_middleware_providers')
 def test_list_deployments(provider):
-    """Tests deployments count between UI, DB and Management system
+    """Tests deployments list between UI, DB and Management system
+    This test requires that no any other provider should exist before.
 
     Steps:
+        * Creates a provider.
         * Get deployments list from UI
         * Get deployments list from Database
         * Get deployments list from Management system(Hawkular)
         * Compare size of all the list [UI, Database, Management system]
     """
-    ui_deps = MiddlewareDeployment.deployments(provider=provider)
-    db_deps = MiddlewareDeployment.deployments_in_db()
-    mgmt_deps = MiddlewareDeployment.deployments_in_mgmt(provider)
-    assert len(ui_deps) == len(db_deps) == len(mgmt_deps), \
-        ("Size of deployments mismatch! UI:{}, DB:{}, MGMT:{}"
-         .format(len(ui_deps), len(db_deps), len(mgmt_deps)))
+    provider.create(cancel=False, validate_credentials=False)
```

Comment:
fixtture evaluation is done by topological search, the search itself starts at autouse fixtures + usefuxtures marker * argnames

the scope is used as well